src/js/libcs/Operators.js

```diff
@@ -3607,30 +3691,143 @@ evaluator.allpoints$0 = function(args, modifs) {
     };
 };
 
+evaluator.allpoints$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "P", csgeo.points);
+};
+
+evaluator.allpoints$1 = function(args, modifs) {
+    return eval_helper.allhelper1arg(args, "P");
+};
+
 evaluator.allmasses$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "Mass", masses);
+};
+
+// collects only points
+evaluator.allmasses$1 = function(args, modifs) {
     var erg = [];
-    for (var i = 0; i < masses.length; i++) {
-        erg[i] = {
-            ctype: "geo",
-            value: masses[i],
-            type: "P"
-        };
+    var pts = eval_helper.allhelper1arg(args, "P");
+    var pp;
+    for (var i = 0; i < pts.value.length; i++) {
+        pp = pts.value[i];
+        if (pp.value.behavior && pp.value.behavior.type === "Mass") erg.push(pp);
     }
     return {
         ctype: "list",
         value: erg
     };
+
+};
+
+evaluator.allsprings$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "Spring", springs);
+};
+
+
+// collects only segments
+evaluator.allsprings$1 = function(args, modifs) {
+    var erg = [];
+    var pts = eval_helper.allhelper1arg(args, "S", "Segment");
+    var sp;
+    for (var i = 0; i < pts.value.length; i++) {
+        sp = pts.value[i];
+        if (sp.value.behavior && sp.value.behavior.type === "Spring") erg.push(sp);
+    }
+    return {
+        ctype: "list",
+        value: erg
+    };
+
 };
 
 evaluator.alllines$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "L", csgeo.lines);
+};
+
+evaluator.alllines$1 = function(args, modifs) {
+    return eval_helper.allhelper1arg(args, "L");
```

Comment:
In most places, CindyJS treats segments as a special case of lines. So `alllines` should return segments, too, and there should be a test case to verify this.